src/sys/timer.cxx

```diff
@@ -46,13 +53,61 @@ double Timer::getTime(const Timer::timer_info& info) {
   return seconds{info.time}.count();
 }
 
+double Timer::getTotalTime(const Timer::timer_info& info) {
+  if (info.running) {
+    return seconds{info.total_time + (clock_type::now() - info.started)}.count();
+  }
+  return seconds{info.total_time}.count();
+}
+
 double Timer::resetTime(Timer::timer_info& info) {
-  auto val = info.time;
+  auto current_duration = info.time;
   info.time = clock_type::duration{0};
   if (info.running) {
-    auto cur_time = clock_type::now();
-    val += cur_time - info.started;
-    info.started = cur_time;
+    const auto current_time = clock_type::now();
+    const auto elapsed = current_time - info.started;
+    current_duration += elapsed;
+    info.started = current_time;
+    info.total_time += elapsed;
+  }
+  return seconds{current_duration}.count();
+}
+
+void Timer::listAllInfo() {
+  const std::string headerName = "Timer name";
+  const std::string headerTime = "Current time (s)";
+  const std::string headerTotal = "Total time (s)";
+  const std::string headerHits = "Hits";
+
+  const auto longest_width = [](const std::string& header, const auto& op) {
```

Comment:
A comment may help here -- for me it's not immediately obvious what each stage of this is doing, although I can guess.

Reply:
Added some comments to explain what's going on